manim/utils/iterables.py

```diff
@@ -33,18 +52,17 @@ def adjacent_n_tuples(objects: Sequence, n: int) -> zip:
 
     Examples
     --------
-    Normal usage::
-
-        list(adjacent_n_tuples([1, 2, 3, 4], 2))
-        # returns [(1, 2), (2, 3), (3, 4), (4, 1)]
+    .. code-block:: pycon
 
-        list(adjacent_n_tuples([1, 2, 3, 4], 3))
-        # returns [(1, 2, 3), (2, 3, 4), (3, 4, 1), (4, 1, 2)]
+        >>> list(adjacent_n_tuples([1, 2, 3, 4], 2))
+        [(1, 2), (2, 3), (3, 4), (4, 1)]
+        >>> list(adjacent_n_tuples([1, 2, 3, 4], 3))
+        [(1, 2, 3), (2, 3, 4), (3, 4, 1), (4, 1, 2)]
     """
     return zip(*([*objects[k:], *objects[:k]] for k in range(n)))
 
 
-def adjacent_pairs(objects: Sequence) -> zip:
+def adjacent_pairs(objects: Sequence[object]) -> zip:
```

Comment:
```suggestion
def adjacent_pairs(objects: Sequence[T]) -> zip[T]:
```

Reply:
Same as previous, `zip` is not subscriptable.